chain-core/src/state/account.rs

```diff
@@ -0,0 +1,115 @@
+use crate::common::{hash256, Timespec, HASH_SIZE_256};
+use crate::init::address::RedeemAddress;
+use crate::init::coin::Coin;
+use blake2::Blake2s;
+use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+use rlp::{Decodable, DecoderError, Encodable, Rlp, RlpStream};
+use std::mem;
+
+/// account state update counter
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
+pub struct Nonce(u64);
```

Comment:
Why a wrapper type instead of `type Nonce = u64;`?

Reply:
for enc/dec + opaque types are less error-prone than type aliases